src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1890,8 +1898,11 @@
     (try
       (store!)
       (catch org.postgresql.util.PSQLException e
-        ;; 42P01 undefined table
-        (if (= "42P01" (.getSQLState e))
+        ;; * 42P01 undefined table -- inheritance based partitions, or
+        ;; declarative partitions when no children yet attached
+        ;; * 23514 check constraint violation -- declarative partitions when row
+        ;; does not match check constraint of existing children partitions
+        (if (or (= "42P01" (.getSQLState e)) (= "23514" (.getSQLState e)))
```

Comment:
A new error test is introduced here because you can get back a 23514 attempting to add a row when declarative partitions are being used. Do we need to version guard this? My feeling was no, but I could be overlooking something.

Reply:
Agreed that no version guard was needed here.